cubids/cli.py

```diff
@@ -121,44 +136,115 @@ def cubids_validate():
 
                 logger.info(" ".join(["Processing subject:", subject]))
                 # create a temporary directory and symlink the data
-                with tempfile.TemporaryDirectory() as tmpdirname:
-                    for fi in files_list:
-
-                        # cut the path down to the subject label
-                        bids_start = fi.find(subject)
-
-                        # maybe it's a single file
-                        if bids_start < 1:
-                            bids_folder = tmpdirname
-                            fi_tmpdir = tmpdirname
-
-                        else:
-                            bids_folder = Path(fi[bids_start:]).parent
-                            fi_tmpdir = tmpdirname + '/' + str(bids_folder)
-
-                        if not os.path.exists(fi_tmpdir):
-                            os.makedirs(fi_tmpdir)
-                        output = fi_tmpdir + '/' + str(Path(fi).name)
-                        shutil.copy2(fi, output)
-
-                    # run the validator
-                    nifti_head = opts.ignore_nifti_headers
-                    subj_consist = opts.ignore_subject_consistency
-                    call = build_validator_call(tmpdirname,
-                                                nifti_head,
-                                                subj_consist)
+                # TMPDIR isn't networked (available on login + exec nodes), so use bids_dir
+                tmpdir = tempfile.TemporaryDirectory(dir=opts.bids_dir, prefix=".")
+                for fi in files_list:
+
+                    # cut the path down to the subject label
+                    bids_start = fi.find(subject)
+
+                    # maybe it's a single file
+                    if bids_start < 1:
+                        bids_folder = tmpdir.name
+                        fi_tmpdir = tmpdir.name
+
+                    else:
+                        bids_folder = Path(fi[bids_start:]).parent
+                        fi_tmpdir = tmpdir.name + '/' + str(bids_folder)
+
+                    if not os.path.exists(fi_tmpdir):
+                        os.makedirs(fi_tmpdir)
+                    output = fi_tmpdir + '/' + str(Path(fi).name)
+                    shutil.copy2(fi, output)
+
+                # run the validator
+                nifti_head = opts.ignore_nifti_headers
+                subj_consist = opts.ignore_subject_consistency
+                call = build_validator_call(tmpdir.name,
+                                            nifti_head,
+                                            subj_consist)
+
+                if queue is False:
```

Comment:
if not queue:

Reply:
an empty queue would evaluate to False but that's probably what we want anyways; changed